src/shared.js

```diff
@@ -19,8 +21,24 @@ function u8toArray(u8) {
   return array;
 }
 
+function validateInteger(value, name) {
+  let err;
+
+  if (typeof value !== 'number')
+    err = new Errors.EINVAL(name, 'number', value);
+
+  if (err) {
+    Error.captureStackTrace(err, validateInteger);
+    throw err;
```

Comment:
I kept the throw instead of returning the error to keep it inline with how Node.js does it - what do you think?

Reply:
Yeah, match node's API and behaviour.